documentation/modules/security/ref-certificates-and-secrets.adoc

```diff
@@ -0,0 +1,151 @@
+// Module included in the following assemblies:
+//
+// assembly-security.adoc
+
+[id='certificates-and-secrets-{context}']
+= Secrets
+
+Strimzi uses _Secrets_ to store private keys and certificates for Kafka cluster components and clients.
+Secrets are used for establishing TLS encrypted connections between Kafka brokers, and between brokers and clients.
+They are also used for mutual TLS authentication.
+
+* A _Cluster Secret_ contains a cluster CA certificate to sign Kafka broker certificates, and is used by a connecting client to establish a TLS encrypted connection with the Kafka cluster to validate broker identity.
+* A _Client Secret_ contains a client CA certificate for a user to sign its own client certificate to allow mutual authentication against the Kafka cluster.
+The broker validates the client identity through the client CA certificate itself.
+* A _User Secret_ contains a private key and certificate, which are generated and signed by the client CA certificate when a new user is created.
+The key and certificate are used for authentication and authorization when accessing the cluster.
+
+Secrets provide private keys and certificates in PEM and PKCS #12 formats.
+Using private keys and certificates in PEM format means that users have to get them from the Secrets, and generate a corresponding truststore (or keystore) to use in their Java applications.
+PKCS #12 storage provides a truststore (or keystore) that can be used directly.
+
+All keys are 2048 bits in size.
+
+CA certificate validity periods are expressed as a number of days after certificate generation.
+You can configure the validity period of:
+
+* Cluster CA certificates in `Kafka.spec.clusterCa.validityDays`
+* Client CA certificates in `Kafka.spec.clientsCa.validityDays`
+
+== PKCS #12 storage
+
+PKCS #12 defines an archive file format (`.p12`) for storing cryptography objects into a single file with password protection.
+You can use PKCS #12 to manage certificates and keys in one place.
+
+Each Secret contains fields specific to PKCS #12.
+
+* The `.p12` field contains the certificates and keys.
+* The `.password` field is the password that protects the archive.
+
+The password is encoded  with base64.
```

Comment:
I do not think it is encoded with base64 apart from the geenral secret encoding. So this should be either removed or written in more detail to explain why do we stress it out (do not copy the password value directly etc.)

In the current state it suggests the password file it self is encoded with base64, not just the secret field.

Reply:
I've removed. Let me know if you think we should drill down into detail instead.